### ISSUE TITLE
[`ruff`] Extend RUF059 to flag unpackings where every target is unused

### ISSUE THREAD

**Issue by Faithfinder** [astral-sh/ruff#27010]
## Summary

When **all** targets of an unpacking assignment are unused, `RUF059`'s per-variable fix produces bindings that are pointless as a whole:

```python
_a, _b = f()          # after RUF059 fix — nothing is ever read
```

Here the assignment conveys nothing. If the call matters only for its side effects, it shouldn't bind a tuple at all:

```python
f()
```

I'd like `RUF059` to detect the "every target unused" case and suggest dropping the assignment.

## Behavior

Trigger when an unpacking has no subsequently-used target:

- `a, b = f()` — neither used
- `_a, _b = f()` — the post-RUF059-fix shape
- `_, _ = f()` — all dummies (currently invisible to RUF059, since every target matches `dummy-variable-rgx`)

Do **not** trigger when at least one target is used (`a, _b = f()` stays as-is).

## Fix

Strip the target list, keep the RHS as an expression statement — safe because side effects are preserved:

```diff
- _a, _b = f()
+ f()
```

Removing the whole line (dead when the RHS is pure) could be an unsafe fix, since purity generally isn't known.

**Comment by ntBre** [comment]
It turns out that this came up when the rule was first discussed: https://github.com/astral-sh/ruff/issues/8884#issuecomment-1868352103.  It's at least not safe to remove the assignment entirely because it could conceal an existing error:

```pycon
>>> def foo(): return 1, 2, 3
...
>>> x, y = foo()
Traceback (most recent call last):
  File "<python-input-1>", line 1, in <module>
    x, y = foo()
    ^^^^
ValueError: too many values to unpack (expected 2, got 3)
>>> foo()
(1, 2, 3)
```

This problem doesn't apply to [unused-variable (F841)](https://docs.astral.sh/ruff/rules/unused-variable/#unused-variable-f841), which explains why F841 is allowed to remove the assignment.

If we had the ability to offer multiple fix options (https://github.com/astral-sh/ruff/issues/2713), I think it would be nice to offer this as one option in an editor context, for example, but I think the current autofix is preferable for CLI usage.
